### PR TITLE
Elasticsearch Particle

### PR DESCRIPTION
**Pull request by Bryantle1028**
Implemented elasticsearch particle with tests in testdata.json and example_es_domain.py

### CODE REVIEW

**Comment by davidyum** [comment]
I think you have a bunch of palcebo json files (96 for describe) from running record test multiple times. Could you just include one of each.

**Comment by brianhlee** [comment]
You have to delete all the useless describes that were returned while the particle was still spinning up. Remember I mentioned that you can just delete them and renumber the ones left over so that it skips over the 10 min spin up time.

David mentioned that there might be an option for placebo to skip the empty responses. Might be worth looking into that if you don't want to do it manually.

File: examples/particle/aws/es/use_es/example_elasticsearch_domain.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#                "Resource": 'arn:aws:es:us-east-1:471176887411:domain/pcf_test_domain/*'
+#                }
+#            ]
+#        }),
+        'SnapshotOptions' : {
+            'AutomatedSnapshotStartHour': 0
+        },
+        'VPCOptions': {
+            'SubnetIds': [
+                'subnet-6bf0d333',
```
Comment (davidyum) [inline examples/particle/aws/es/use_es/example_elasticsearch_domain.py:44]: can you remove subnets. Use dummy values like subnet-1111111

File: examples/particle/aws/es/use_es/example_elasticsearch_domain.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                'subnet-3738020a'
+            ],
+            'SecurityGroupIds': [
+                'sg-2d991c57'
```
Comment (davidyum) [inline examples/particle/aws/es/use_es/example_elasticsearch_domain.py:48]: can you remove this too

File: examples/particle/aws/es/use_es/example_elasticsearch_domain.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            'Enabled': True
+        },
+        'Tags': [
+            {
+                'Key': 'ASV',
+                'Value': 'ASVCLOUDCUSTODIAN'
+            },
+            {
+                'Key': 'OwnerContact',
+                'Value': 'janitorialservices@capitalone.com'
```
Comment (davidyum) [inline examples/particle/aws/es/use_es/example_elasticsearch_domain.py:64]: same with tags

File: examples/particle/aws/es/use_es/example_elasticsearch_domain.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+es_particle = ESDomain(es_example_json)
+
+#example start
+es_particle.set_desired_state(State.running)
+es_particle.apply()
+
+es_example_json['aws_resource']['SnapshotOptions']['AutomatedSnapshotStartHour'] = 1
+es_example_json['aws_resource']['Tags'][1]['Value'] = 'zhb257'
```
Comment (davidyum) [inline examples/particle/aws/es/use_es/example_elasticsearch_domain.py:81]: same here

File: pcf/particle/aws/es/es_domain.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+from botocore.errorfactory import ClientError
+#TEST
+import itertools
+import logging
+
+logger = logging.getLogger(__name__)
+
+class ESDomain(AWSResource):
+    """
+    This is the implementation of Amazon's S3 service.
```
Comment (davidyum) [inline pcf/particle/aws/es/es_domain.py:27]: change s3 to elasticsearch

File: pcf/particle/aws/es/es_domain.py [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+                tag_set.append({
+                    "Key": k,
+                    "Value": v
+                })
+
+            self.client.add_tags(ARN=self._arn, TagList=tag_set)
+
+    def _stop(self):
+        """
+        Elasticsearch does not have a sotpped state so it calls terminate.
```
Comment (davidyum) [inline pcf/particle/aws/es/es_domain.py:96]: typo for stopped

**Review by davidyum — CHANGES_REQUESTED** [review]

**Comment by Bryantle1028** [comment]
> I think you have a bunch of palcebo json files (96 for describe) from running record test multiple times. Could you just include one of each.

**Comment by Bryantle1028** [comment]
> I think you have a bunch of palcebo json files (96 for describe) from running record test multiple times. Could you just include one of each.

Should I just include one of each call even if some are not empty?

**Comment by brianhlee** [comment]
Just delete everything in the folder and run generate once. Then go through and find the chunk of empty describe calls that are made while the particle is spinning up. This will make the test run faster without waiting the 10 mins it takes to spin it up.

**Pull request closed without merging**